test/dotnet-new3.UnitTests/AllProjectsWork.cs

```diff
@@ -1,78 +1,95 @@
-//        {
-//            string workingDir = GetWorkingDirectoryName(testName);
-//            Directory.CreateDirectory(workingDir);
+        private static string GetWorkingDirectoryName(string suffix, [CallerMemberName] string callerName = "")
+        {
+            return Path.Combine(Path.GetTempPath(), _tempDirParent.ToString(), callerName, suffix);
+        }
 
-//            Program.Main(new[] { "--debug:reinit" });
-            
-//            if (string.IsNullOrEmpty(Environment.GetEnvironmentVariable("DN3")))
-//            {
-//                string relTo = new Uri(typeof(Program).GetTypeInfo().Assembly.CodeBase, UriKind.Absolute).LocalPath;
-//                relTo = Path.GetDirectoryName(relTo);
-//                relTo = Path.Combine(relTo, @"..\..\..\..\..\dev");
-//                Environment.SetEnvironmentVariable("DN3", relTo);
-//            }
+        [Theory]
+        [InlineData("emptyweb_cs-50", "Microsoft.DotNet.Web.ProjectTemplates.5.0", "web")]
+        [InlineData("mvc_cs-50", "Microsoft.DotNet.Web.ProjectTemplates.5.0", "mvc")]
+        [InlineData("mvc_fs-50", "Microsoft.DotNet.Web.ProjectTemplates.5.0", "mvc", "-lang", "F#")]
+        [InlineData("api_cs-50", "Microsoft.DotNet.Web.ProjectTemplates.5.0", "api")]
+        [InlineData("emptyweb_cs-31", "Microsoft.DotNet.Web.ProjectTemplates.3.1", "web", "-f", "netcoreapp3.1")]
+        [InlineData("mvc_cs-31", "Microsoft.DotNet.Web.ProjectTemplates.3.1", "mvc", "-f", "netcoreapp3.1")]
+        [InlineData("mvc_fs-31", "Microsoft.DotNet.Web.ProjectTemplates.3.1", "mvc", "-lang", "F#", "-f", "netcoreapp3.1")]
+        [InlineData("api_cs-31", "Microsoft.DotNet.Web.ProjectTemplates.3.1", "api", "-f", "netcoreapp3.1")]
+        [InlineData("console_cs-31", null, "console", "-f", "netcoreapp3.1")]
+        [InlineData("library_cs-50", null, "library", "-f", "net5.0")]
+        public void AllWebProjectsRestoreAndBuild(string testName, string installNuget, params string[] args)
+        {
+            string workingDir = GetWorkingDirectoryName(testName);
+            Directory.CreateDirectory(workingDir);
 
-//            CommandResult result = Command.Create("dotnet-new3", args, outputPath: Environment.GetEnvironmentVariable("DN3"))
-//                .WorkingDirectory(workingDir)
-//                .CaptureStdErr()
-//                .CaptureStdOut()
-//                .Execute();
-//            Assert.Equal(0, result.ExitCode);
+            Program.Main(new[] { "--debug:reinit" });
 
-//            result = Command.CreateDotNet("restore", new string[0])
-//                .WorkingDirectory(workingDir)
-//                .CaptureStdErr()
-//                .CaptureStdOut()
-//                .Execute();
-//            Assert.Equal(string.Empty, result.StdErr);
-//            Assert.Equal(0, result.ExitCode);
+            if (string.IsNullOrEmpty(Environment.GetEnvironmentVariable("DN3")))
```

Comment:
Should we use more unique pattern for variable? 
Should we remove it after unit tests are run?